airflow/providers/amazon/aws/utils/__init__.py

```diff
@@ -21,13 +21,17 @@
 from datetime import datetime
 from enum import Enum
 
+from deprecated import deprecated
+
+from airflow.utils.helpers import prune_dict
 from airflow.version import version
 
 log = logging.getLogger(__name__)
 
 
+@deprecated(reason="use prune_dict() instead")
 def trim_none_values(obj: dict):
-    return {key: val for key, val in obj.items() if val is not None}
+    return prune_dict(obj)
```

Comment:
In providers we raise `AirflowProviderDeprecationWarning`.

this means our static checks are not good enough as they don't catch this case
https://github.com/apache/airflow/blob/d6a4f851e0a0c3d99b327b7fca23fa002a915df7/.pre-commit-config.yaml#L431

Reply:
hmm ok I didn't know that. I have actually used this annotation before. Maybe we could have our own annotation to make it easier to write deprecation warnings.